### PR TITLE
Feature/autotoc

### LINKED ISSUE #6: Auto TOC
We could add a GH action for this  
https://github.com/ekalinin/github-markdown-toc

### PR DESCRIPTION
**Pull request by carmelom**
Closes #6

### CODE REVIEW

File: .github/workflows/markdown-toc.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      - README.md
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - uses: actions/checkout@v2
+      - run: |
+          curl https://raw.githubusercontent.com/ekalinin/github-markdown-toc/master/gh-md-toc -o gh-md-toc
```
Comment (wipfli) [inline .github/workflows/markdown-toc.yml:18]: Rather than downloading `master` we should get the file at some specific commit.

Comment (wipfli) [reply .github/workflows/markdown-toc.yml:18]: Even better would be to use an action from the marketplace I guess.

File: .github/workflows/markdown-toc.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+          chmod a+x gh-md-toc
+          ./gh-md-toc --insert --no-backup README.md
+      - uses: stefanzweifel/git-auto-commit-action@v4
+        with:
+          commit_message: Auto update markdown TOC
```
Comment (wipfli) [inline .github/workflows/markdown-toc.yml:23]: add `file_pattern: README.md`

File: gh-md-toc [hunk 3]
```diff
@@ -0,0 +1,361 @@
+#!/usr/bin/env bash
```
Comment (wipfli) [inline gh-md-toc:1]: Delete file.

**Comment by wipfli** [comment]
Thanks for this pull request. It will make maintaining a lot easier I guess. Regarding the action trigger I think you can actually remove it on push to main because the action will already run on the pull request and add the table of contents there, like for example in 646622c.

File: .github/workflows/markdown-toc.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+name: Markdown TOC
+
+on:
+  push:
+    paths:
+      - README.md
```
Comment (wipfli) [inline .github/workflows/markdown-toc.yml:6]: Remove, table of content will be generated and committed on pull request trigger.

Comment (carmelom) [reply .github/workflows/markdown-toc.yml:6]: But we also want to auto-update it when new commits are pushed directly to the main repo.

Comment (wipfli) [reply .github/workflows/markdown-toc.yml:6]: `main` is protected. We will always go through pull requests I suggest.

**Comment by carmelom** [comment]
TBH, we can just use the one rendered by github:

https://github.blog/changelog/2021-04-13-table-of-contents-support-in-markdown-files/

But I kinda like having the TOC in the readme itself. What do you think about it?

File: .github/workflows/markdown-toc.yml (hunk 1 above)
Comment (carmelom) [reply .github/workflows/markdown-toc.yml:18]: I don't think I could find one on the marketplace equivalent to this one, perhaps the author just didn't want to publish it (yet)...  this is the template for the action I found on the repo itself: https://github.com/ekalinin/github-markdown-toc#toc-generation-with-github-actions.

**Comment by wipfli** [comment]
Uh that table of contents looks actually much better. I like it!

File: .github/workflows/markdown-toc.yml (hunk 1 above)
Comment (wipfli) [reply .github/workflows/markdown-toc.yml:18]: Don't worry. `https://raw.githubusercontent.com/ekalinin/github-markdown-toc/17ec017/gh-md-toc` should do.

**Comment by wipfli** [comment]
> But I kinda like having the TOC in the readme itself.

I like it too and it is also the [recommended best practice](https://github.com/sindresorhus/awesome/blob/main/awesome.md).

**Comment by wipfli** [comment]
It is strange that the commit action fails. Maybe you could swap it out with https://github.com/marketplace/actions/add-commit ?

**Comment by wipfli** [comment]
Well, or maybe just do this (https://github.com/stefanzweifel/git-auto-commit-action#checkout-the-correct-branch):

```yml
- uses: actions/checkout@v2
  with:
    ref: ${{ github.head_ref }}
```

**Pull request closed without merging**